src/app/wallets/ln-send-payment.ts

```diff
@@ -306,35 +306,32 @@ const executePaymentViaIntraledger = async ({
   const recipientWallet = await WalletsRepository().findById(recipientWalletId)
   if (recipientWallet instanceof Error) return recipientWallet
 
-  const lnFee = toSats(0)
-  const sats = toSats(amount + lnFee)
-  const usd = sats * usdPerSat
-  const usdFee = lnFee * usdPerSat
+  const amountDisplayCurrency = DisplayCurrencyConversionRate(usdPerSat)(amount)
 
   return LockService().lockWalletId(
     { walletId: senderWallet.id, logger },
     async (lock) => {
       const balance = await LedgerService().getWalletBalance(senderWallet.id)
       if (balance instanceof Error) return balance
-      if (balance < sats) {
+      if (balance < amount) {
         return new InsufficientBalanceError(
-          `Payment amount '${sats}' exceeds balance '${balance}'`,
+          `Payment amount '${amount}' exceeds balance '${balance}'`,
         )
       }
 
       const journal = await LockService().extendLock({ logger, lock }, async () =>
         LedgerService().addLnIntraledgerTxSend({
           senderWalletId: senderWallet.id,
+          senderWalletCurrency: senderWallet.currency,
+          senderUsername: senderAccount.username,
           paymentHash,
           description,
-          sats,
-          fee: lnFee,
-          usd,
-          usdFee,
+          sats: amount,
+          amountDisplayCurrency,
           recipientWalletId,
-          pubkey: lndService.defaultPubkey(),
-          senderUsername: senderAccount.username,
+          recipientWalletCurrency: recipientWallet.currency,
           recipientUsername: null,
+          pubkey: lndService.defaultPubkey(),
```

Comment:
Just thinking about this again, the `defaultPubkey` may not be relevant here because there's no send happening and there's no guarantee that it's the same as the recipient invoice's destination pubkey. 

It might be more relevant to use the `recipientPubkey` value at this point (for when we eventually move to multiple lnds).

Reply:
just changing the order of property here to group sender/recipient- not changing the logic on the pubkey. is there are some things to improve it, would you mind creating a PR for it?

Reply:
Yup it's a separate thing, I can PR that